### ISSUE TITLE
systemd-login fails to start when dbus-broker is enabled

### ISSUE THREAD

**Issue by kamiyaa** [bus1/dbus-broker#33]
I had a similar problem when messagebus group was not in /etc/group and /etc/passwd (not from dbus-broker)
This is on Gentoo, Linux-4.9.44, Systemd-234, dbus-1.10.22
with dbus-broker commit cf08097127c2e20058c157e51dd5ec665f73412f

```
....
systemd-logind[439]: Failed to add match for JobRemoved: Connection timed out
systemd-logind[439]: Failed to fully start up daemon: Connection timed out
systemd[1]: Failed to subscribe to NameOwnerChanged signal for 'org.freedesktop.login1': Connection timed out
systemd[1]: Failed to subscribe to NameOwnerChanged signal for 'org.freedesktop.NetworkManager': Connection timed out
systemd[1]: Failed to subscribe to activation signal: Connection timed out
systemd[1]: Failed to register name: Connection timed out
systemd[1]: Failed to set up API bus: Connection timed out
....
systemd[1]: systemd-logind.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: Failed to start Login Service.
systemd[1]: systemd-logind.service: Unit entered failed state.
systemd[1]: systemd-logind.service: Failed with result 'exit-code'.
....
```

What I did:
```
~ # systemctl enable dbus-broker.service
~ # systemctl reboot
```

**Comment by dvdhrm** [comment]
These log-messages look like the fallout after a bus failure. Are you sure there is no other log-message *before* those that tells you about some dbus-broker errors?

Also, what do you mean with this happens only if "messagebus" is not in /etc?

**Comment by kamiyaa** [comment]
> Also, what do you mean with this happens only if "messagebus" is not in /etc?
Never mind, its just missing messagebus in /etc/group and /etc/passwd causes dbus to fail.

Also, I believe this is the culprit:
```
....
systemd[1]: Starting Create Volatile Files and Directories...
systemd[1]: Started Create Volatile Files and Directories.
systemd[1]: Starting Update UTMP about System Boot/Shutdown...
systemd[1]: Started Update UTMP about System Boot/Shutdown.
systemd[1]: Reached target System Initialization.
systemd[1]: Listening on D-Bus System Message Bus Socket.
systemd[1]: Reached target Sockets.
systemd[1]: Reached target Basic System.
systemd[1]: Reached target Containers.
systemd[1]: Started D-Bus System Message Bus.
dbus-broker-launch[401]: Listening on inherited socket
dbus-broker-launch[401]: ERROR manager_run_child @ ../src/launch/main.c +302: No such file or directory
dbus-broker-launch[401]: Loading service '/usr/share/dbus-1/system-services/org.freedesktop.UDisks2.service'
dbus-broker-launch[401]: ERROR manager_load_service @ ../src/launch/main.c +653: Connection reset by peer
dbus-broker-launch[401]:       manager_load_services @ ../src/launch/main.c +714
dbus-broker-launch[401]:       manager_run @ ../src/launch/main.c +856
dbus-broker-launch[401]:       run @ ../src/launch/main.c +1021
dbus-broker-launch[401]:       main @ ../src/launch/main.c +1054
dbus-broker-launch[401]: Exiting due to fatal error: -104
systemd[1]: Failed to subscribe to NameOwnerChanged signal for 'org.freedesktop.login1': Connection timed out
systemd[1]: Failed to subscribe to NameOwnerChanged signal for 'org.freedesktop.NetworkManager': Connection timed out
systemd[1]: Failed to subscribe to activation signal: Connection timed out
systemd[1]: Failed to register name: Connection timed out
systemd[1]: Failed to set up API bus: Connection timed out
systemd[1]: Starting Login Service...
....
systemd[1]: dbus-broker.service: Main process exited, code=exited, status=1/FAILURE
systemd[1]: dbus-broker.service: Unit entered failed state.
systemd[1]: dbus-broker.service: Failed with result 'exit-code'.
....
```

**Comment by dvdhrm** [comment]
Yes, this error message means the launcher could not find the binary of `dbus-broker`. I assume you compiled the project manually? I recommend using `--prefix /usr` when running `meson`.

I will leave this bug open, since we really ought to pass the prefix into the launcher so we don't rely on the binary residing in `/usr`.

**Comment by kamiyaa** [comment]
Thanks!

I can boot into my machine now and from what I can tell so far, everything works as before.

**Comment by teg** [comment]
This was fixed in PR #44. Closing.
